code/part-two/client/source/App.jsx

```diff
@@ -1,22 +1,86 @@
 import React from 'react';
 import { Link, Route, Switch } from 'react-router-dom';
 
+import { SignupLogin } from './SignupLogin';
+
 import { Collection } from './Collection';
 import { CollectionList } from './CollectionList';
 import { Moji } from './Moji';
 import { Offer } from './Offer';
 import { OfferList } from './OfferList';
 
 export class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      privateKey: this.props.privateKey,
+      publicKey: this.props.publicKey
+    }
+  }
+
+  set privateKey(key) {
+    this.setState({ privateKey: key });
+    localStorage.setItem('privateKey', key);
+  }
+
+  get privateKey() {
+    return this.state.privateKey 
+      ? this.state.privateKey
+      : localStorage.getItem('privateKey');
+  }
+
+  set publicKey(key) {
+    this.setState({ publicKey: key });
+    localStorage.setItem('publicKey', key);
+  }
+
+  get publicKey() {
+    return this.state.publicKey 
+      ? this.state.publicKey
+      : localStorage.getItem('publicKey');
+  }
+
+  logout() {
+    this.privateKey = null;
+    this.publicKey = null;
+    localStorage.removeItem('privateKey');
+    localStorage.removeItem('publicKey');
+  }
+  
   render() {
+    const publicKey = this.publicKey
     return (
       <div>
         <nav>
           <Link to="/">Home</Link>&ensp;
+          <Link to="/signup-login">Sign Up/Login</Link>&ensp;
           <Link to="/collection">View Collections</Link>&ensp;
           <Link to="/offer">View Offers</Link>&ensp;
+
+          {/* public key view / logout */}
```

Comment:
Is this empty object literal with a comment inside a React thing? Should this Just be a `//` comment?

Reply:
@delventhalz yeah, this is how you write comments in JSX

the `{ }` isn't an empty object literal; it's escaping the JSX (allowing you to write JavaScript)